libs/dotnet-sdk/Registration/RegisterFeatureBoard.cs

```diff
@@ -0,0 +1,76 @@
+using Featureboard.DotnetSdk.Models;
+using Featureboard.DotnetSdk.State;
+using Featureboard.DotnetSdk.UpdateStrategies;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Options;
+
+namespace Featureboard.DotnetSdk.Registration;
+
+public static class RegisterFeatureBoard
+{
+  private static UpdateStrategy updateStrategy = UpdateStrategy.None;
+  public static FeatureBoardBuilder AddFeatureBoard<TFeatures, TAudienceProvider>(this IServiceCollection services) where TFeatures : class, IFeatures where TAudienceProvider : class, IAudienceProvider
+  {
+    var configuration = services.BuildServiceProvider().GetRequiredService<IConfiguration>();
+    services.AddOptions<FeatureboardOptions>().Bind(configuration.GetSection(nameof(FeatureboardOptions))).ValidateDataAnnotations();
+
+    services.AddHttpClient(Constants.HttpClientName, (serviceProvider, client) =>
```

Comment:
If going with my suggestion of taking direct dependency on `HttpClient` in `FeatureboardHttpClient` I think the registration method to use here needs to change to the following:
```
services.AddHttpClient<IFeatureboardHttpClient, FeatureboardHttpClient>().ConfigureHttpClient((serviceProvider, client) =>
```

Reply:
I need to read further through the docs when implementing.